rs-e2e/src/main/java/gov/hhs/cdc/trustedintermediary/rse2e/hl7/HL7Encoding.java

```diff
@@ -0,0 +1,78 @@
+package gov.hhs.cdc.trustedintermediary.rse2e.hl7;
+
+import java.util.Map;
+
+/** The HL7Encoding class represents the encoding characters used in an HL7 message. */
+public record HL7Encoding(Map<String, Character> characters) {
+    public static final String DEFAULT_SEGMENT_DELIMITER = "\n";
+    public static final char DEFAULT_FIELD_DELIMITER = '|';
+    public static final char DEFAULT_COMPONENT_DELIMITER = '^';
+    public static final char DEFAULT_REPETITION_DELIMITER = '~';
+    public static final char DEFAULT_ESCAPE_CHARACTER = '\\';
```

Comment:
Is there a reason the default escape character is not '\'. I'm assuming we need to escape the escape character for some reason?

Reply:
Right, we need to escape the escape character. That's because `\` is also an escape character in Java

Reply:
Will the `\n` also need a second `\` to escape its `\`? It would also be helpful to add a clarifying comment about the `\\`

Reply:
The `\` in `\n` is the escape character itself so we don't need another one. This is all standard Java syntax so not sure if we need to add further clarification. If you write a Windows path string in Java you would write `C:\\Program Files\\Java`, and if you write a newline in a string you'd write `Hello\nWorld`

Reply:
Ugh Java is so weird, thanks for the explanation